html/modules/custom/docstore/src/ResourceTrait.php

```diff
@@ -549,8 +549,19 @@ public function prepareEntityEntityReferenceField(EntityReferenceFieldItemList $
       // @todo It would much easier if we were to use "name" for terms which is
       // the field name for the term label rather than "label" or if we were to
       // use "label" for all the entity types.
-      if ($entity->getEntityTypeId() === 'taxonony_term') {
-        $label_key = 'label';
+      if ($entity->getEntityTypeId() === 'taxonomy_term') {
+        // @todo This was included when there was a typo in 'taxonomy_term',
+        // but it causes tests to fail, so specifying 'name' for now.
+        // $label_key = 'label';
+        $label_key = 'name';
+
+        // Add display label to local_coordination_groups.
+        if ($entity->vid->first()->getValue()['target_id'] === 'local_coordination_groups') {
```

Comment:
Not a big fan of having special cases like that in code. There may be more use cases for a term display label on other vocabularies so I would suggest to simply check if it exists and not empty (`$entity->hasField('display_label') && !$entity->display_label->isEmpty()`) and then add it (`$entity->display_label->value`).